src/plugins/event-plugins/PageViewPlugin.ts

```diff
@@ -74,27 +91,41 @@ export class PageViewPlugin extends MonkeyPatched implements Plugin {
                 url?: string | null
             ): void {
                 const retVal = original.apply(this, arguments);
-                self.recordPageView();
+                self.recordRouteChange();
                 return retVal;
             };
         };
     };
 
+    /**
+     * Checks the current browser's URL against the current Page object's id.
+     * If they are different, indicate route change has occurred, start Page creation.
+     */
+    private recordRouteChange = () => {
+        const newUrl = location.pathname + location.hash;
+        if (this.context.getCurrentUrl() !== newUrl) {
```

Comment:
Agreed, now that the feature will not rely on `PATH_AND_HASH` to distinguish unique pages, it will use the existing `recordPageView` method in PageManager to decide whether to create a new page or not.